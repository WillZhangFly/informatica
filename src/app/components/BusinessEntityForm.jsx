import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import TextInput from "./TextInput";

function BusinessEntityForm() {
  let originValue = useSelector((state) => state.entity);

  const [entity, setEntity] = useState(useSelector((state) => state.entity));
  const [saveResult, setSaveResult] = useState({});
  const normalizeMeta = useSelector((state) => {
    return state.meta.field.filter((_metaField) => entity[_metaField.name]);
  });

  console.log("compare", originValue, entity);

  const saveEntity = (e) => {
    e.preventDefault();
    setSaveResult({
      ...entity,
      $original: {
        ...originValue,
      },
    });
  };
  const handleInputChange = (e) => {
    setEntity({
      ...entity,
      [e.target.id]: e.target.value,
    });
  };

  return (
    <div className="justify-content-center d-flex">
      <div className="card p-3 col-6 mt-5">
        <h2>Business Entity</h2>
        <form onSubmit={saveEntity}>
          <div className="form-group row">
            {normalizeMeta.map((_field) => {
              return (
                <TextInput
                  handleChange={handleInputChange}
                  value={entity[_field.name]}
                  key={_field.name}
                  name={_field.label}
                  id={_field.name}
                  dataType={_field.dataType}
                />
              );
            })}
          </div>
          <button
            type="submit"
            className="form-control mt-2 btn btn-primary col-4"
          >
            Save
          </button>
          <div className="mt-4">
            <h2>Saved results print here:</h2>
            {JSON.stringify(saveResult)};
          </div>
        </form>
      </div>
    </div>
  );
}

export default BusinessEntityForm;
