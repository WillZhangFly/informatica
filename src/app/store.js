import { configureStore } from "@reduxjs/toolkit";
import entityReducer from "../features/entity/entitySlice";
import logger from "redux-logger";

export default configureStore({
  reducer: {
    entity: entityReducer,
  },
  middleware: [logger],
});
