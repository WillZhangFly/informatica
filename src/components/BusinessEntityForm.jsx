import React from "react";

function BusinessEntityForm() {
  return (
    <div className="justify-content-center d-flex">
      <div className="card p-3 col-6 mt-5">
        <h2>Business Entity</h2>
        <form onSubmit={() => {}}>
          <div class="form-group row">
            <label for="firstName" class="col-sm-2 col-form-label">
              First Name
            </label>
            <div class="col-sm-4">
              <input type="text" class="form-control" id="firstName" />
            </div>
            <label for="lastName" class="col-sm-2 col-form-label">
              Last Name
            </label>
            <div class="col-sm-4">
              <input type="text" class="form-control" id="lastName" />
            </div>
          </div>
          <button type="submit" className="form-control mt-2 btn btn-primary">
            Save
          </button>
        </form>
      </div>
    </div>
  );
}

export default BusinessEntityForm;
