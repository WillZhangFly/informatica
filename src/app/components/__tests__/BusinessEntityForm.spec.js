import React from "react";
import { shallow } from "enzyme";
import BusinessEntityForm from "../BusinessEntityForm";
import { Provider } from "react-redux";
import store from "../../../app/store";

describe("<BusinessEntityForm />", () => {
  it("should render", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <BusinessEntityForm />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
