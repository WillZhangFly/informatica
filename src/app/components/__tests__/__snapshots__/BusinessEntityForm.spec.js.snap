// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessEntityForm /> should render 1`] = `ShallowWrapper {}`;
